internal/eval/eval.go

```diff
@@ -1493,3 +1495,38 @@ func evalGEODIST(args []string, store *dstore.Store) []byte {
 
 	return clientio.Encode(utils.RoundToDecimals(result, 4), false)
 }
+
+// evalRANDOMKEY returns a random key from the currently selected database.
+func evalRANDOMKEY(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) > 0 {
+		return makeEvalError(diceerrors.ErrWrongArgumentCount("RANDOMKEY"))
+	}
+
+	availKeys, err := store.Keys("*")
+	if err != nil {
+		return makeEvalError(diceerrors.ErrGeneral("could not get keys"))
+	}
+
+	if len(availKeys) > 0 {
+		for range len(availKeys) {
+			randKeyIdx, err := rand.Int(rand.Reader, big.NewInt(int64(len(availKeys))))
+			if err != nil {
+				continue
+			}
+
+			randKey := availKeys[randKeyIdx.Uint64()]
+			keyObj := store.Get(randKey)
+			if keyObj == nil {
+				continue
+			}
+
+			currTimeMs := uint64(utils.GetCurrentTime().UnixMilli())
+			expireTimeMs, isExpirySet := dstore.GetExpiry(keyObj, store)
+			if (isExpirySet && expireTimeMs > currTimeMs) || !isExpirySet {
+				return makeEvalResult(clientio.Encode(randKey, false))
+			}
```

Comment:
This should not be needed, since `store.Get` returns nil already if the key has expired.